tools/rc/usResourceCompiler.cpp

```diff
@@ -532,107 +531,74 @@ int main(int argc, char** argv)
   // ---------------------------------------------------------------------------------
 
   // Create a new zip archive which will be copied to zipfile later
-  dbg_print("Creating zip archive\n");
-  remove(zipFile);
+  std::clog << "Creating zip archive " << outfile.c_str() << std::endl;
+  remove(outfile.c_str());
 
-  if (!mz_zip_writer_init_file(&writeArchive, zipFile, 0))
+  if (!mz_zip_writer_init_file(&writeArchive, outfile.c_str(), 0))
   {
     exit_printf(&writeArchive, "Internal error, could not init new zip archive\n");
   }
-  dbg_print("Initialized zip archive\n");
+  std::clog << "Initialized zip archive" << std::endl;
 
-  // Add current files to the zip archive
-  zipArgIndices = malloc_or_abort(argc * sizeof *zipArgIndices);
-  while(++argIndex < argc)
+  // Add resource files to the zip archive
+  for (option::Option* opt = options[RESFILE]; opt; opt = opt->next())
   {
-    const char* fileName = argv[argIndex];
-    const size_t fileNameLength = strlen(fileName);
-
-    // determine the argument type
-    if (strcmp(argv[argIndex], "-a") == 0)
-    {
-      mergeFlag = 0;
-      continue;
+    if (!bundleName.size()) {
```

Comment:
Couldn't this check for the bundle name be moved farther up? It seems redundant to check the bundle name each time a resource is added to the zip file.